### PR TITLE
Enhancement memory usage for PyTorch quantization

### PR DESCRIPTION
**Pull request by PenghuiCheng**
Signed-off-by: Cheng, Penghui <penghui.cheng@intel.com>

## Type of Change

Enhancement memory usage for PyTorch quantization.  
API changed

## Description

detail description 
JIRA ticket: [2668](https://jira.devtools.intel.com/browse/ILITV-2668)

## Expected Behavior & Potential Risk

Memory usage be reduced when model be quantized in PyTorch with performance only mode. 

## How has this PR been tested?

Local test

### CODE REVIEW

File: neural_compressor/adaptor/pytorch.py [hunk 1]
```diff
@@ -2922,7 +2924,7 @@ def quantize(self, tune_cfg, model, dataloader, q_func=None):
 
         # PyTorch 1.13 and above version, need example_inputs for fx trace, but it not realy used,
         # so set it to None.
-        example_inputs = None
+        self.example_inputs = None
```
Comment (PenghuiCheng) [inline neural_compressor/adaptor/pytorch.py:2927]: remove it?

File: neural_compressor/adaptor/pytorch.py [hunk 2]
```diff
@@ -3148,7 +3150,7 @@ def _pre_hook_for_qat(self, dataloader=None):
...
```
Comment (PenghuiCheng) [inline neural_compressor/adaptor/pytorch.py:3153]: remove it?

**Review by xin3he — APPROVED** [review]

File: neural_compressor/adaptor/pytorch.py (hunk 2 above)
Comment (xin3he) [reply neural_compressor/adaptor/pytorch.py:3153]: For fine-grianed fx, the example_inputs for each module maybe a big issue, so I keep it to notice us that example_inputs is not used right now.

File: neural_compressor/adaptor/pytorch.py (hunk 1 above)
Comment (xin3he) [reply neural_compressor/adaptor/pytorch.py:2927]: For fine-grianed fx, the example_inputs for each module maybe a big issue, so I keep it to notice us that example_inputs is not used right now.

File: test/ipex/test_adaptor_ipex.py [hunk 3]
```diff
@@ -151,5 +152,17 @@ def test_bf16(self):
...
+        from neural_compressor.experimental import Quantization
+        model = M()
+        config.quantization.example_inputs = torch.randn([1, 3, 224, 224])
+        quantizer = Quantization(config)
+        quantizer.model = model
+        quantizer.conf.usr_cfg.tuning.exit_policy['performance_only'] = False
+        dataset = quantizer.dataset('dummy', (100, 3, 224, 224), label=True)
+        dataloader = torch.utils.data.DataLoader(dataset)
+        quantizer.calib_dataloader = dataloader
+        nc_model = quantizer.fit()
```
Comment (chensuyue) [inline test/ipex/test_adaptor_ipex.py:165]: assert required

**Pull request merged**